fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java

```diff
@@ -168,6 +169,18 @@ private void publishVersionForOlapTable(List<TransactionState> readyTransactionS
                     }
                     // clear publish version tasks to reduce memory usage when state changed to visible.
                     transactionState.clearPublishVersionTasks();
+
+                    // Refresh materialized view when base table update transaction has been visible
+                    long dbId = transactionState.getDbId();
+                    Database db = GlobalStateMgr.getCurrentState().getLocalMetastore().getDb(dbId);
+                    for (long tableId : transactionState.getTableIdList()) {
+                        Table table = db.getTable(tableId);
+                        Set<Long> relatedMvs = ((OlapTable) table).getRelatedMaterializedViews();
+                        for (long mvId : relatedMvs) {
```

Comment:
should refresh only load-triggered refresh strategy mvs, not all related mvs here.